File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    # 투 포인터
    max_sum = 0
    end = 0
    interval = len(sequence)

    for start in range(len(sequence)):
        while max_sum < k and end < len(sequence):
            max_sum += sequence[end]
            end += 1
        if max_sum == k and end-1-start < interval:
            res = [start, end-1]
            interval = end-1-start
        max_sum -= sequence[start]

    return res

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    let answer = [];
    
    // 투 포인터
    let endPointer = 0;
    let sequenceSum = 0;
    let sequenceLength = sequence.length;
    let interval = sequence.length;
    
    for (let startPointer = 0; startPointer < sequenceLength; startPointer++) {
        while (endPointer < sequenceLength && sequenceSum < k) {
            sequenceSum += sequence[endPointer];
            endPointer += 1;
        }
        
        if (sequenceSum == k && endPointer-1-startPointer < interval) {
            answer = [startPointer, endPointer-1];
            interval = endPointer - 1 - startPointer;
        }
        
        sequenceSum -= sequence[startPointer];
    }
    
    return answer;
}